src/graph.jl

```diff
@@ -56,9 +56,14 @@ end
 
 "Returns `true` if `g` is a `DiGraph`."
 is_directed(g::Graph) = false
-has_edge(g::Graph, e::Edge) = (e in edges(g)) || (reverse(e) in edges(g))
+has_edge(g::Graph, e::Edge) = isordered(e) ? e in edges(g) : reverse(e) in edges(g)
+
+isordered(e::Edge) = src(e) <= dst(e)
```

Comment:
Since we're now ensuring that `src <= dst` for all edges in Graph, would you mind rewriting `==(g::Graph, h::Graph)` to get rid of the current horrible inefficiency? Thanks.